posts/2023-09-05-Increasing-Apple-Version-Requirements.md

```diff
@@ -0,0 +1,79 @@
+
+### tvOS
+
+The third generation AppleTV (released 2012-2013) is no longer supported.
+
+# Why are the requirements being changed?
+
+Prior to now, Rust claimed support for very old Apple OS versions, but many never even received passive testing or support. This is a rough place to be for a toolchain, as it hinders opportunities for improvement in exchange for a support level many people, or everyone, will never utilize. For Apple's mobile platforms, many of the old versions are now even unable to receive new software due to App Store publishing restrictions.
+
+Additionally, the past two years have clearly indicated that Apple, the sole controller of toolchains for these targets, are making it difficult-to-impossible to support them anymore. As of XCode 14, last year's toolchain release, building for many old OS versions [became unsupported](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes). XCode 15 continues this. After enough time, continuing to use an older toolchain can even lead to breaking build issues for others.
+
+We want Rust to be a first-class option for developing software for and on Apple's platforms, but to continue this goal we have to set an easier, and more realistic compatibility baseline. The new requirements were determined after surveying what Apple and third-party statistics are available to us and picking a middle ground that balances compatibility with Rusts's needs and limitations.
+
+# Do I need to do anything?
+
+If you or an application you develop are affected by this change, there are different options which may be helpful:
+- If possible, raise your minimum supported OS versions. All OS versions discussed in above have no support from the vendor. Not even security updates.
+-  If you are running the Rust compiler or other host tools that were previously supported, consider cross-compiling from a newer host instead. You may also no longer be able to depend on the Rust standard library.
+- If none of these options work, you may need to freeze the version of the Rust toolchain your project builds with. Alternatively, you may be able to maintain a custom toolchain that supports your requirements for any sub-component of it.
+
+If your project does not directly support a specific version, but instead depends on a default previously used by Rust, there are some steps you can take
+to help improve. For example, a number of crates in the ecosystem have hardcoded Rust's default support versions since they haven't changed for a long time:
+- If you use the `cc` crate to include build languages into your project, a [future update](https://github.com/rust-lang/cc-rs/pull/848) will handle this transparently.
+- If you need a minimum OS version for anything else, crates should query the new `rustc --print deployment-target` option for a default, or user-set, value 
+on newer toolchains going forward. Hardcoded defaults should only be used for older toolchains where this is unavailable.
```

Comment:
Maybe specify the first rust version in which `--print deployment-target` was stabilized? It seems to be stable on current stable at least.

Reply:
It insta-stabilized in 1.71, but I'll add that since its a good call.